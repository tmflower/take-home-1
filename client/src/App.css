body {
   background-image: url("../public/background.jpg");
   background-size: cover;
   color: white;
   font-family: 'Varela Round', sans-serif;
}

a:link {
    color: white;
    text-decoration: none;
}

a:visited {
    color: white;
    text-decoration: none;
}

a:hover {
    color: white;
    text-decoration: none;
    scale: 1.5;
}

a:active {
    color: white;
    text-decoration: none;
}
