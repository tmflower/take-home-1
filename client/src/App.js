import './App.css';
import { Routes, Route } from "react-router-dom";
import { Navbar } from "./Navbar";
import { Home } from "./routes/Home";
import { Signup } from "./routes/Signup";
import { Admin } from "./routes/Admin";
import { useState, useEffect } from "react";
import axios from 'axios';

function App() {
  const [users, setUsers] = useState([]);
  const [userAdded, setUserAdded] = useState(false);

  useEffect(() => {
    async function getUsers() {
      const usersList = await axios.get("http://localhost:3001/users");
      setUsers(usersList.data);
    }
    getUsers();
  }, [userAdded]);

  return (
    <div>
    <Navbar></Navbar>
    <Routes>
      <Route path="/" element={<Home />}></Route>
      <Route path="/signup" element={<Signup setUserAdded={setUserAdded}/>}></Route>
      <Route path="/admin" element={<Admin users={users}/>}></Route>
    </Routes>
    </div>

  );
}

export default App;
