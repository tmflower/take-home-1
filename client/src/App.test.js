import { render, screen } from '@testing-library/react';
import { App } from './App';
import { MemoryRouter } from 'react-router-dom';

test('renders witihout crashing', () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>);
});

test('matches snapshot', () => {
  const { asFragment } =
  render(
      <MemoryRouter>
          <App />
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

jest.mock('axios');
test('displays landing page', async() => {
  render(
      <MemoryRouter>
          <App />
      </MemoryRouter>
  );
  const h1 = screen.getByText('Welcome to Yodlr!');
  expect(h1).toBeInTheDocument();
});
