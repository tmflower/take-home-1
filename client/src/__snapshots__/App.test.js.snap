// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <div
        class="MuiBox-root css-0"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-xc68p4-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1x5wx18-MuiToolbar-root"
          >
            <a
              aria-current="page"
              class="link active"
              element="[object Object]"
              href="/"
            >
              Home
            </a>
            <a
              class=""
              element="[object Object]"
              href="/signup"
            >
              Signup
            </a>
            <a
              class=""
              data-testid="adminLink"
              element="[object Object]"
              href="/admin"
            >
              Admin
            </a>
          </div>
        </header>
      </div>
    </nav>
    <div
      class="MuiBox-root css-1d27sqt"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-kmsxfs-MuiTypography-root"
      >
        Welcome to Yodlr!
      </h1>
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1xx1ags-MuiTypography-root"
      >
        Who are you?
      </h3>
      <div
        class="MuiBox-root css-1xhj18k"
      >
        <a
          class="link"
          href="/signup"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1b50r3g-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            User
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          class=""
          href="/admin"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1b50r3g-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Admin
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
